src/features/runtime-checks.js

```diff
@@ -157,32 +162,64 @@ class DDGRuntimeChecks extends HTMLElement {
 
         let prepend = ''
         const aggregatedLookup = new Map()
+        let currentScope = null
         /* Convert the config into a map of scopePath -> { key: value } */
         for (const [key, value] of Object.entries(processedConfig)) {
             const path = key.split('.')
-            const scopePath = path.slice(0, -1).join('.')
+
+            currentScope = aggregatedLookup
             const pathOut = path[path.length - 1]
-            if (aggregatedLookup.has(scopePath)) {
-                aggregatedLookup.get(scopePath)[pathOut] = value
-            } else {
-                aggregatedLookup.set(scopePath, {
-                    [pathOut]: value
-                })
-            }
+            // Traverse the path and create the nested objects
+            path.slice(0, -1).forEach((pathPart, index) => {
+                if (!currentScope.has(pathPart)) {
+                    currentScope.set(pathPart, new Map())
+                }
+                currentScope = currentScope.get(pathPart)
+            })
+            currentScope.set(pathOut, value)
         }
 
-        for (const [key, value] of aggregatedLookup) {
-            const path = key.split('.')
-            if (path.length !== 1) {
-                console.error('Invalid config, currently only one layer depth is supported')
-                continue
+        /**
+         * Output scope variable definitions to arbitrary depth
+         */
+        function stringifyScope (scope, scopePath) {
+            let output = ''
+            for (const [key, value] of scope) {
+                const varOutName = [...scopePath, key].join('_')
+                if (value instanceof Map) {
+                    const proxyName = `_proxyFor_${varOutName}`
+                    output += `
+                    let ${proxyName}
+                    if (${scopePath.join('?.')}?.${key} === undefined) {
```

Comment:
with this not be transpiled, I'm assuming you're ok with the support level of optional chaining?

Reply:
👀 checks notes

Reply:
I think we're good based on the pixels I'm seeing.